test/scripts/products/h22-validate-that-rate-limit-service-is-working-as-expected/h22-validate-that-rate-limit-service-is-working-as-expected.go

```diff
@@ -236,33 +252,51 @@ func cleanupRedis(ctx context.Context, client k8sclient.Client, namespace string
 // and an error if an unexpected error occurred when performing the test
 func testCountIncreases(r *redisCounter, endpoint string, numberOfRequests int) (bool, error) {
 	// call endpoint for rate limiting key to show in redis
-	requestEndpoint(endpoint)
+	for {
+		err, code := requestEndpoint(endpoint)
+		if err != nil && code != http.StatusNotFound && code != http.StatusForbidden {
+			fmt.Printf("Error on initial request of endpoint: %s\n", err)
+			break
+		}
+		if code == http.StatusOK {
+			break
+		}
+		if testRunStart.Add(endpointTimeout).Before(time.Now()) {
+			fmt.Println("Timeout waiting for the endpoint to become functional")
+			break
+		}
+		fmt.Println("  ☁️  Waiting for the endpoint to become functional")
+		time.Sleep(10 * time.Second)
+	}
 
-	initialCount, err := r.GetCount()
+	initialCount, err := getCount(r, time.Now())
 	if err != nil {
 		return false, err
 	}
 
 	var wg sync.WaitGroup
 
 	for i := 0; i < numberOfRequests; i++ {
+		wg.Add(1)
 		go func() {
+			defer wg.Done()
 			requestEndpoint(endpoint)
-			wg.Done()
 		}()
-		wg.Add(1)
 	}
-
 	wg.Wait()
+	// time.Sleep(30 * time.Second)
```

Comment:
Should this commented out line be removed? If not, could you add a line explaining why it's needed